pytorch_lightning/trainer/trainer.py

```diff
@@ -2056,10 +2056,12 @@ def devices(self) -> int:
         return self.num_devices
 
     @property
-    def data_parallel_device_ids(self) -> Optional[List[int]]:
-        return (
-            self._accelerator_connector.parallel_device_ids if self._accelerator_connector.parallel_device_ids else None
+    def data_parallel_device_ids(self) -> List[int]:
+        rank_zero_deprecation(
+            "`Trainer.data_parallel_device_ids` was deprecated in v1.6 and will be removed in v1.8."
+            " Please use `Trainer.device_ids` instead."
         )
+        return self.device_ids if isinstance(self.accelerator, GPUAccelerator) else []
```

Comment:
why doesn't this always return `self.device_ids` ? why is the GPU accelerator check needed?

Reply:
I think `data_parallel_device_ids` is originally intended to return the ids of GPU devices not device_ids? 

https://github.com/PyTorchLightning/pytorch-lightning/blob/f2d6a855af0c130f81e2c04506ba03a061eba924/pytorch_lightning/trainer/connectors/accelerator_connector.py#L836-L838.

Also its existing usage were all related to GPUs.